doc/dev/docstring_typehint.md

```diff
@@ -0,0 +1,185 @@
+* [Type Hints](#type_hints)
+    * [Type Hints for Python 2.7 and 3.5+](#type_hints_for_python_2.7_and_3.5+)
+    * [Type Hints for Python 3.5+](#type_hints_for_python_3.5+)
+
+## Docstrings
+
+Docstrings are noted by the Python long-string `"""<docstring>"""`. When adding docstrings the `~` can be optionally added to the front of a custom model, this will only display the actual model name in the documentation. For example a type that looks like `azure.data.tables.TableEntity` will display the entire path in the documentation, but `~azure.data.tables.TableEntity` will only display `TableEntity` in the documentation. Additionally, adding the wrapper `` :class:`~azure.data.tables.TableEntity` `` will display only `TableEntity` formatted in monospace font. Both of these capabilities are optional and can only be used in the two-line format described below.
+
+### Method Docstrings
+
+A method docstring is annotated by the Python long-string `"""<docstring>"""` right after the method definition. Below is an example of a full docstring from `azure-ai-formrecognizer`:
+```python
+    @distributed_trace
+    def begin_training(self, training_files_url, use_training_labels, **kwargs):
+        # type: (str, bool, Any) -> LROPoller[CustomFormModel]
+        """Create and train a custom model. The request must include a `training_files_url` parameter that is an
+        externally accessible Azure storage blob container URI (preferably a Shared Access Signature URI). Note that
+        a container URI (without SAS) is accepted only when the container is public.
+        Models are trained using documents that are of the following content type - 'application/pdf',
+        'image/jpeg', 'image/png', 'image/tiff', or 'image/bmp'. Other types of content in the container is ignored.
+
+        :param str training_files_url: An Azure Storage blob container's SAS URI. A container URI (without SAS)
+            can be used if the container is public. For more information on setting up a training data set, see:
+            https://docs.microsoft.com/azure/cognitive-services/form-recognizer/build-training-data-set
+        :param bool use_training_labels: Whether to train with labels or not. Corresponding labeled files must
+            exist in the blob container if set to `True`.
+        :keyword str prefix: A case-sensitive prefix string to filter documents in the source path for
+            training. For example, when using a Azure storage blob URI, use the prefix to restrict sub
+            folders for training.
+        :keyword bool include_subfolders: A flag to indicate if subfolders within the set of prefix folders
+            will also need to be included when searching for content to be preprocessed. Not supported if
+            training with labels.
+        :keyword str model_name: An optional, user-defined name to associate with your model.
+        :keyword str continuation_token: A continuation token to restart a poller from a saved state.
```

Comment:
We should have an example with inline type and clarify that for `keyword` you then need to use `paramtype`, while a regular `param` should be `type`

Reply:
You say that later, but still this overview should show it

Reply:
Half use inline type, half use paramtype.